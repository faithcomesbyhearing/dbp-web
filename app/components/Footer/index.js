/**
 *
 * Footer
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import SvgWrapper from '../SvgWrapper';

function Footer({
	settingsActive,
	profileActive,
	searchActive,
	notebookActive,
	toggleNotebook,
	toggleSettingsModal,
	toggleProfile,
	toggleSearch,
	isScrollingDown,
}) {
  return (
<div
		className={
			isScrollingDown ? 'footer-background closed' : 'footer-background'
		}
>
		<div className="footer-content">
			<span
				title={'Profile'}
				id={'profile-button'}
				className={profileActive ? 'item active' : 'item'}
				role="button"
				tabIndex={0}
				onClick={() => !profileActive && toggleProfile()}
			>
				<SvgWrapper className={'icon'} fill="#fff" svgid="profile" />
			</span>
			<span
				title={'Search'}
				id={'search-button'}
				className={searchActive ? 'item active' : 'item'}
				role="button"
				tabIndex={0}
				onClick={() => !searchActive && toggleSearch()}
			>
				<SvgWrapper className={'icon'} fill="#fff" svgid="search" />
			</span>
			<span
				title={'Notebook'}
				id={'notebook-button'}
				className={notebookActive ? 'item active' : 'item'}
				role="button"
				tabIndex={0}
				onClick={() => !notebookActive && toggleNotebook()}
			>
				<SvgWrapper className={'icon'} fill="#fff" svgid="notebook" />
			</span>
			<span
				title={'Settings'}
				id={'settings-button'}
				className={settingsActive ? 'item active' : 'item'}
				role="button"
				tabIndex={0}
				onClick={() => !settingsActive && toggleSettingsModal()}
			>
				<SvgWrapper className={'icon'} fill="#fff" svgid="text_options" />
			</span>
		</div>
</div>
);
}

Footer.propTypes = {
	settingsActive: PropTypes.bool,
	profileActive: PropTypes.bool,
	searchActive: PropTypes.bool,
	notebookActive: PropTypes.bool,
	isScrollingDown: PropTypes.bool,
	toggleNotebook: PropTypes.func,
	toggleSettingsModal: PropTypes.func,
	toggleProfile: PropTypes.func,
	toggleSearch: PropTypes.func,
};

export default Footer;
