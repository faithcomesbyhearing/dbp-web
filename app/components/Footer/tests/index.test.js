import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Footer from '..';

const props = {
  settingsActive: false,
  profileActive: false,
  searchActive: false,
  notebookActive: false,
  isScrollingDown: false,
  toggleNotebook: jest.fn(),
  toggleSettingsModal: jest.fn(),
  toggleProfile: jest.fn(),
  toggleSearch: jest.fn(),
};

describe('Footer component tests', () => {
  beforeEach(() => {
    // Reset mock functions before each test
    props.toggleNotebook.mockClear();
    props.toggleSettingsModal.mockClear();
    props.toggleProfile.mockClear();
    props.toggleSearch.mockClear();
  });

  it('matches snapshot when open', () => {
    const { container } = render(<Footer {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot when closed', () => {
    const { container } = render(<Footer {...props} isScrollingDown />);
    expect(container).toMatchSnapshot();
  });

  it('should toggle each setting when clicked', () => {
    render(<Footer {...props} />);

    // Simulate profile button click
    fireEvent.click(screen.getByTitle('Profile'));
    expect(props.toggleProfile).toHaveBeenCalledTimes(1);

    // Simulate search button click
    fireEvent.click(screen.getByTitle('Search'));
    expect(props.toggleSearch).toHaveBeenCalledTimes(1);

    // Simulate notebook button click
    fireEvent.click(screen.getByTitle('Notebook'));
    expect(props.toggleNotebook).toHaveBeenCalledTimes(1);

    // Simulate settings button click
    fireEvent.click(screen.getByTitle('Settings'));
    expect(props.toggleSettingsModal).toHaveBeenCalledTimes(1);
  });
});
