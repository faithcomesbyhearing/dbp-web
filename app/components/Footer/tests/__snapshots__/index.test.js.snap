// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component tests matches snapshot when closed 1`] = `
<div
  className="footer-background closed"
>
  <div
    className="footer-content"
  >
    <span
      className="item"
      id="profile-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
      title="Profile"
    >
      <svg
        className="icon"
        fill="#fff"
      >
        <use
          xlinkHref="/svglist.svg#profile"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
    <span
      className="item"
      id="search-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
      title="Search"
    >
      <svg
        className="icon"
        fill="#fff"
      >
        <use
          xlinkHref="/svglist.svg#search"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
    <span
      className="item"
      id="notebook-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
      title="Notebook"
    >
      <svg
        className="icon"
        fill="#fff"
      >
        <use
          xlinkHref="/svglist.svg#notebook"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
    <span
      className="item"
      id="settings-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
      title="Settings"
    >
      <svg
        className="icon"
        fill="#fff"
      >
        <use
          xlinkHref="/svglist.svg#text_options"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`Footer component tests matches snapshot when open 1`] = `
<div
  className="footer-background"
>
  <div
    className="footer-content"
  >
    <span
      className="item"
      id="profile-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
      title="Profile"
    >
      <svg
        className="icon"
        fill="#fff"
      >
        <use
          xlinkHref="/svglist.svg#profile"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
    <span
      className="item"
      id="search-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
      title="Search"
    >
      <svg
        className="icon"
        fill="#fff"
      >
        <use
          xlinkHref="/svglist.svg#search"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
    <span
      className="item"
      id="notebook-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
      title="Notebook"
    >
      <svg
        className="icon"
        fill="#fff"
      >
        <use
          xlinkHref="/svglist.svg#notebook"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
    <span
      className="item"
      id="settings-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
      title="Settings"
    >
      <svg
        className="icon"
        fill="#fff"
      >
        <use
          xlinkHref="/svglist.svg#text_options"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </div>
</div>
`;
